packages/remix-server-runtime/server.ts

```diff
@@ -1019,6 +1043,39 @@ async function assertResponsesMatch(_a: Response, _b: Response) {
   }
 }
 
+function stripStackTraceFromDefaultErrorBoundary(text: string) {
+  let openPreStart = text.indexOf("<pre ");
+  let openPreEnd = text.indexOf('">', openPreStart + 1);
+  let closePre = text.indexOf("</pre>", openPreEnd + 1);
+  let error = text.substring(openPreEnd + 2, closePre);
+  let stackStart = error.indexOf("\n");
+  let errorWithoutStack = error.substring(0, stackStart);
+  return (
+    text.replace(error, "ERROR REMOVED") + "\n\nError:\n" + errorWithoutStack
+  );
+}
+
+function stripStackTraceFromRemixContext(text: string) {
+  let scriptStart = text.indexOf("<script>window.__remixContext");
+  let scriptEnd = text.indexOf("</script>", scriptStart + 1);
+  let scriptContents = text.substring(scriptStart, scriptEnd + 9);
+  let remixContext = JSON.parse(
+    scriptContents
+      .replace("<script>window.__remixContext = ", "")
+      .replace(";</script>", "")
+  );
+  // If we didn't have an error - assert the full document
+  if (!remixContext?.appState?.error?.stack) {
+    return text;
+  }
+
+  // Otherwise remove the stack trace and assert the text/JSON as a
+  // concatenated string
+  remixContext.appState.error.stack = "yes";
+  text = text.replace(scriptContents, "CONTEXT REMOVED");
+  return text + "\n\nContext:\n" + JSON.stringify(remixContext);
```

Comment:
Here we end up asserting the full HTML without `<script>window.__remixContext = {...}</script>` concatenated with the stringifed `remixContext` with `stack` replaced